### DISCUSSION TITLE
posts/2022/07/pagefind-quite-find-site-search/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#23] [Announcements]
# posts/2022/07/pagefind-quite-find-site-search/

It used to be that having search on a static site was a hassle — and perhaps an expensive one — but Pagefind has changed all that.

https://www.brycewray.com/posts/2022/07/pagefind-quite-find-site-search/

**Comment by regisphilibert** [comment]
Amazing thanks!

Nobody is recommending it so maybe I'm missing something... But for dev mode in Hugo I would recommend publishing once to the `static` directory and then run `hugo server` like you always do. This way you only need to build pagefind locally if new content is critical to your current task.

All you have to do is add `static/_pagefind` to your `.gitignore` and it's 👌

**Reply by regisphilibert** [reply]
Turns out you can... `--bundle-dir` has to be below `public`...

**Reply by regisphilibert** [reply]
yes and here's the way: `npx -y pagefind --source public --bundle-dir ../static/_pagefind`

**Reply by brycewray** [reply]
@regisphilibert As always, sir, you figure out these things before everyone else! 😃 Thanks for that info.

cc: @bglw

**Reply by rodrigoalcarazdelaosa** [reply]
Came here to say that I was looking for a way to make this work with `hugo server` but @regisphilibert already found a way. Would you @brycewray mind adding his recommendation to your post? Thanks a lot!

**Reply by brycewray** [reply]
@rodrigoalcarazdelaosa Good idea. Will do.

**Reply by brycewray** [reply]
@regisphilibert @rodrigoalcarazdelaosa
Now updated: https://www.brycewray.com/posts/2022/07/pagefind-quite-find-site-search/#update-2021-07-31 — thanks again!
